examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,361 @@
+    super.deactivate();
+  }
+
+  @override
+  void didUpdateWidget(FadeAnimation oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.child != widget.child) {
+      animationController.forward(from: 0.0);
+    }
+  }
+
+  @override
+  void dispose() {
+    animationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return animationController.isAnimating
+        ? new Opacity(
+            opacity: 1.0 - animationController.value,
+            child: widget.child,
+          )
+        : new Container();
+  }
+}
+
+// Shows a
+class ConnectivityOverlay extends StatefulWidget {
+  final Widget child;
+  final Completer<Null> connectedCompleter;
+  final GlobalKey<ScaffoldState> scaffoldKey;
+
+  const ConnectivityOverlay(
+      {this.child, this.connectedCompleter, this.scaffoldKey});
+
+  @override
+  _ConnectivityOverlayState createState() => new _ConnectivityOverlayState();
+}
+
+class _ConnectivityOverlayState extends State<ConnectivityOverlay> {
+  StreamSubscription<ConnectivityResult> connectivitySubscription;
+  bool connected = true;
+
+  static const Widget errorSnackBar = const SnackBar(
+    backgroundColor: Colors.red,
+    content: const ListTile(
+      title: const Text('No network'),
+      subtitle: const Text(
+          'To load the videos you must have an active network connection'),
+    ),
+  );
+
+  Stream<ConnectivityResult> connectivityStream() async* {
+    final Connectivity connectivity = new Connectivity();
+    ConnectivityResult previousResult = await connectivity.checkConnectivity();
+    yield previousResult;
+    await for (ConnectivityResult result
+        in connectivity.onConnectivityChanged) {
+      if (result != previousResult) {
+        yield result;
+        previousResult = result;
+      }
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    connectivitySubscription =
+        connectivityStream().listen((ConnectivityResult connectivityResult) {
+      print(connectivityResult);
+      if (!mounted) {
+        return;
+      }
+      if (connectivityResult == ConnectivityResult.none) {
+        widget.scaffoldKey.currentState.showSnackBar(errorSnackBar);
+      } else {
+        if (!widget.connectedCompleter.isCompleted) {
+          widget.connectedCompleter.complete(null);
+        }
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    connectivitySubscription.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) => widget.child;
+}
+
+class VideoDemo extends StatefulWidget {
+  const VideoDemo({Key key}) : super(key: key);
+
+  static const String routeName = '/video';
+
+  @override
+  _VideoDemoState createState() => new _VideoDemoState();
+}
+
+class _VideoDemoState extends State<VideoDemo>
+    with SingleTickerProviderStateMixin {
+  final VideoPlayerController butterflyController =
+      new VideoPlayerController(butterflyUri);
```

Comment:
Trailing comma. Though this one is a little annoying.
```dart
  final VideoPlayerController butterflyController = new VideoPlayerController(
    butterflyUri,
  );
```

Reply:
done